monitor/monitor.go

```diff
@@ -111,12 +107,58 @@ func (m *Monitor) handleConnection(server net.Listener) {
 		}
 
 		mutex.Lock()
+		first := listeners.Len() == 0
 		listeners.PushBack(conn)
 		log.WithField("count.listener", listeners.Len()).Info("New monitor connected.")
+		/* Since this is our first listener that means the event loop
+		 * is not running and has to be started.
+		 */
+		if first {
+			go m.mainEventLoop()
+		}
 		mutex.Unlock()
 	}
 }
 
+// mainEventLoop is responsible for reading events from the perf ring buffer
+// and manage the connections to the socket
+func (m *Monitor) mainEventLoop() {
+	log.Info("Starting monitoring traffic")
+	me, err := bpf.NewPerCpuEvents(eventConfig)
+	if err != nil {
+		log.WithError(err).Error("Error while starting monitor")
+		return
+	}
+	monitorEvents = me
```

Comment:
A nit, but you could assign monitorEvents directly on line 128 (I suspect this is a holdover from a complicated if statement that didn't create monitorEvents in the outer scope).